happiness-backend/tests/test_journals.py

```diff
@@ -92,4 +113,58 @@ def test_journals(init_client):
 
     get = client.get('/api/journal/', headers=auth_key_header(token, user.derive_pwd_key('test2')))
     assert get.status_code == 200
-    assert get.json[1]['data'] == 'secret' and get.json[0]['data'] == 'secret2'
\ No newline at end of file
+    assert get.json[1]['data'] == 'secret' and get.json[0]['data'] == 'secret2'
+
+def test_reset_pwd_recovery_get(init_client):
+    client, token, user = init_client
+    pwd_key = user.derive_pwd_key('test').decode()
+    client.post('/api/journal/', json={'data': 'secret'}, headers=auth_key_header(token, pwd_key))
+    client.post('/api/journal/', json={'data': 'secret2'}, headers=auth_key_header(token, pwd_key))
+
+    add_recovery = client.put('/api/user/info/',
+                            json={'data_type': 'key_recovery_phrase', 'data': 'help'},
+                            headers=auth_key_header(token, pwd_key))
+    assert add_recovery.status_code == 200
+
+    reset_token = generate_confirmation_token('test@example.app')
+
+    bad_reset = client.post('/api/user/reset_password/' + reset_token,
+                                 json={'password': 'W password', 'recovery_phrase': 'nope'},
+                                 headers=auth_key_header(token, None))
+    assert bad_reset.status_code == 400
+
+    reset_password = client.post('/api/user/reset_password/' + reset_token,
+                                json={'password': 'W password', 'recovery_phrase': 'help'},
+                                headers=auth_key_header(token, None))
+    assert reset_password.status_code == 204
+
+    bad_get = client.get('/api/journal/', headers=auth_key_header(token, pwd_key))
+    assert bad_get.status_code == 400
+
+    get = client.get('/api/journal/', headers=auth_key_header(token, user.derive_pwd_key('W password')))
+    assert get.status_code == 200
+    assert get.json[1]['data'] == 'secret' and get.json[0]['data'] == 'secret2'
+
+def test_edit_delete(init_client):
```

Comment:
I think these should be separated into `test_edit` and `test_delete` just to improve readability and have the code as loosely coupled as possible.

Reply:
you should have @Alex967CW fix his too then....